### ISSUE TITLE
342ms regex compile in mozilla.components.support.utils.WebURLFinder

### ISSUE THREAD

**Issue by acreskeyMoz** [mozilla-mobile/android-components#5880]
When starting to type into the URL bar, a long regex compile is seen on the main Java thread.

342ms via the Gecko Profiler timeline on Pixel 3:
https://perfht.ml/2Sg0THi

```
kotlin.text.Regex.<init>()
mozilla.components.support.utils.WebURLFinder.<clinit>()
org.mozilla.fenix.utils.ClipboardHandler.getUrl()
org.mozilla.fenix.search.SearchFragment.onResume()
androidx.fragment.app.Fragment.performResume()
androidx.fragment.app.FragmentManager.moveToState()
androidx.fragment.app.FragmentManager.moveFragmentToExpectedState()
androidx.fragment.app.FragmentManager.moveToState()
androidx.fragment.app.FragmentManager.executeOpsTogether()
...
```

(I downloaded Firefox Preview from the PlayStore, Feb 10).

This looks to be related to this issue:
https://github.com/mozilla-mobile/android-components/pull/5384

Perhaps this can be done before the user interacts with the URL bar?

**Comment by mcomella** [comment]
Can we remove nhe `WebURLFinder` class to prevent this from happening ever again?

**Comment by csadilek** [comment]
@acreskeyMoz Thanks for filing. This was already fixed by Will in https://github.com/mozilla-mobile/android-components/issues/5594.

It looks like you tested this on "Firefox Preview" (not Firefox Preview Nightly) which doesn't have the fix yet (it's landed in A-C 29.0.0 and the latest stable Preview uses A-C 28.0.x).

@mcomella This didn't happen again :). There just wasn't a release since. There's a follow up to remove the strict URL checking: https://github.com/mozilla-mobile/android-components/issues/5680

I am closing this, but please re-open if you still see this. I just verified on master that the slow regex is no longer compiled when typing the URL bar.

**Comment by acreskeyMoz** [comment]
I'm not seeing this issue on master either, thanks for looking into it.
